### ISSUE TITLE
Consider removing the recommendation to set `aria-label` on non-significant sections

### ISSUE THREAD

**Issue by rdeltour** [daisy/kb#4]
By default, a `section` is only mapped as a region landmark when it has an accessible name. It means that if the `aria-label` attribute is set, it will be mapped, when a `section` with a heading will not be mapped by default (see the region mapping in [CORE-AAM](https://w3c.github.io/core-aam/#details-id-50))

I think we should only recommend labeling significant sections (those which have an explicit landmark role).

**Comment by mattgarrish** [comment]
What is an insignificant section, though?

I think it would be better to recommend that people use aria-labelledby with h1-h6 than try to pick out when a untitled section might not need a title. Including aria-label is a useful way to identify quizzes, learning objectives and other content in textbooks that is often only identified by boxes and shading. The reader most likely is going to want these in the list of landmarks.

**Comment by rdeltour** [comment]
I don’t know. Landmarks overuse can create confusing by bloating the landmark navigation (as reminded in this [article by Steve Faulkner](https://developer.paciellogroup.com/blog/2015/09/easy-content-organisation-with-html5/)), especially where navigation by heading is already available. I don’t think we should recommend making *every* h1-h6 section its own landmark.

To me, labeling should be reserved to:

- `section`s that are already identified as landmarks using one of the DPUB ARIA `landmark` roles (e.g. `doc-chapter`, `doc-part`, etc)
- `section`s that contain “content that is sufficiently important for users to be able to navigate to the section” (wording from [ARIA authoring practices](https://w3c.github.io/aria-practices/#aria_lh_region)), which would have an ARIA `region` role.

To summarize, the responsibility of identifying which section is “significant” is left to the author.

**Comment by mattgarrish** [comment]
It's not the most apt comparison, though, as publications are not typically cluttered in the same way as web pages. Add chunking, and we don't get a lot of overload in any given page. These were considerations in adding the landmark roles that we did, and the clutter issue factored into the decisions we made.

Not every section needs to be a landmark, I agree with you there, but the ones that are least significant tend to be in the front and back matter, but those are also ones that are typically titled. Within the body, it's a harder call. If the untitled sections were in the table of contents, then that would be a different story, but the problem is that they often aren't, and without a heading there is no easy way to get to them if they aren't landmarks.

Maybe some guidance about the reasons not to overdo landmarks is all we should offer? "Sufficiently important" is so open to interpretation it's almost meaningless in the context of a publication. Some people will think every section is sufficiently important.

**Comment by rdeltour** [comment]
> It's not the most apt comparison, though, as publications are not typically cluttered in the same way as web pages. Add chunking, and we don't get a lot of overload in any given page. These were considerations in adding the landmark roles that we did, and the clutter issue factored into the decisions we made.

Right, but taken from the other side: in the context of the entire publication, every landmark adds up, so having every single section being marked-up as a landmark could clutter the navigation. I’m aware that no AT expose the full landmarks list, but still, conceptually, it would be weird to say "this is a landmark in the context of this single doc, but shouldn’t be in the landmark nav when considering the entire publication".

> Not every section needs to be a landmark, I agree with you there, but the ones that are least significant tend to be in the front and back matter, but those are also ones that are typically titled. Within the body, it's a harder call. If the untitled sections were in the table of contents, then that would be a different story, but the problem is that they often aren't, and without a heading there is no easy way to get to them if they aren't landmarks.

I agree, and that’s my main point: not every section needs to be a landmark, especially not those that are part of the main outline and are already reachable through the toc nav.

YMMV when considering what’s a _significan_ landmark, but right now our guidelines kind of recommend to make _any_ headless section a landmark, and I don’t think that’s right. It’s more subtle.

> 
> Maybe some guidance about the reasons not to overdo landmarks is all we should offer? 

Yes, some guidance about not overdoing landmarks is good. And some clarification about when a section will appear in AT as landmark and when not could help too.

**Comment by mattgarrish** [comment]
> so having every single section being marked-up as a landmark could clutter the navigation

Where did you see that? That would be excessive, and it should be removed. The guidance was adapted from epub:type, so it may be a conversion error of the documentation. I was only saying above that we need to get aria-label(ledby) used when a landmark is being set, which means one of the existing roles.

We only have landmarks for parts and chapters in the body, as I recall, so there will be lots of subsections not in the landmarks, and nothing wrong with that. We shouldn't go so far as to tell people not to create other landmarks, though, as there will be legitimate uses for which there aren't existing semantics.

The untitled case is the most complicated, because the uses I can think of tend to be important ones. They also tend to be very small in number. If we had an education module, then there probably would be roles for most of them, too.

**Comment by rdeltour** [comment]
> > so having every single section being marked-up as a landmark could clutter the navigation
> 
> Where did you see that?

We don’t say this explicitly, but at the same time I think we’re not clear enough on how and where sections need to be marked-up as landmarks and when not.

Some of the weak spots to improve IMO in the [Sections topic of the KB](http://kb.daisy.org/publishing/docs/html/sections.html):

> Use the `section` element for primary sections of content.

⇒ unless this `section` explicitly has a DPUB landmark role, or is explicitly labelled with `aria-label`(`led-by`), this would have no accessibility support. 

> Example 2 — Untitled section

One could read that as a recommended approach for **all** the untitled section. However, it should **only** be used for the ones that are actually to appear as landmarks. Otherwise, we may end up in a kinda unbalanced state where some (untitled) sections are effective landmarks and some (heading) sections of the same level of importance aren’t.

**Comment by mattgarrish** [comment]
Closing as this also looks like it's been addressed in the faq section for roles.
